packages/core/src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -77,16 +77,19 @@ export function throwIfCancelled() {
     }
 }
 
-export async function getOpenProjects() {
+export async function getOpenProjects(onProjectFirstOpen = false) {
     const folders = vscode.workspace.workspaceFolders
     if (folders === undefined) {
-        void vscode.window.showErrorMessage(
-            CodeWhispererConstants.noSupportedJavaProjectsFoundMessage.replace(
-                'LINK_HERE',
-                CodeWhispererConstants.linkToPrerequisites
+        if (!onProjectFirstOpen) {
+            // only show notification when user invokes Transform, not when validating projects silently
+            void vscode.window.showErrorMessage(
+                CodeWhispererConstants.noSupportedJavaProjectsFoundMessage.replace(
+                    'LINK_HERE',
+                    CodeWhispererConstants.linkToPrerequisites
+                )
             )
-        )
-        throw new ToolkitError('No Java projects found since no projects are open', { code: 'NoOpenProjects' })
+        }
+        throw new ToolkitError('', { name: 'NoProjectsOpen' })
```

Comment:
Throwing an error with an empty message is pretty suspicious. Would be preferred to remove the other, redundant error, and use ToolkitError consistently.